CASMPET-7153 sbps full soln doc

aasha-hpe opened the pull request:
# Description
This is for documentation on iSCSI based boot content projection solution (SBPS) end to end. This has all the details about SBPS solutio, Node personalisation, BOS & COS support for SBPS, GOSS test suite and LIO metrics. 

# Checklist

- [x] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [x] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
````diff
@@ -0,0 +1,198 @@
+iSCSI initiator includes standard Linux kernel, user space iSCSI initiator services and DM (Device Mapper) multi-pathing software. DNS SRV and A records are used to discover the target nodes during the boot and are part of BOS (Boot orchestration service) session template boot parameters. ThisBOS session template is used to rigger the boot of initiator node(s). The LUNs created on the target node ...
+
+Example output snippet of 'targetcli ls' command on worker node where iSCSI LUNS are created for the images scanned:
+```bash
````
This isn't a bash script, it is example output, so leave off `bash`.  Calling it a bash script is causing all the pipes(|) to get displayed in red.

aasha-hpe replied:
done

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
````diff
@@ -0,0 +1,198 @@
+In above 'targetcli ls' command output, we can see that four fileio backing store is created for four rootfs images and four iSCSI LUNs created which has the 'rootfs' id being mapped and ready for rootfs image projection. 
+
+Sample initiator node snippet after image projection:
+```bash
````
Again, leave off the `bash`, this isn't a bash script, it is command output and calling it a bash script is causing all the pipes(|) to be displayed in red.

aasha-hpe replied:
done

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
````diff
@@ -0,0 +1,198 @@
+cray ims images update bbe0e9eb-fa8f-4896-9f54-95dbd26de9bb --metadata-operation set --metadata-key sbps-project --metadata-value true
+```
+Command to describe the IMS image metadata:
+```bash
+cray ims images describe < Image id > --format json
````
Same thing, remove the `bash`, and change `< Image id >` to `<image_id>`
Also close the quote at this point.

aasha-hpe replied:
done

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
````diff
@@ -0,0 +1,198 @@
+cray ims images update < Image id > --metadata-operation set --metadata-key < key > --metadata-value < value >
+```
+e.g.
+```bash
````
Remove the `bash` so that `set` and `true` don't get displayed in blue.

aasha-hpe replied:
done

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,198 @@
+cray ims images describe < Image id > --format json
+e.g
+
+# cray ims images describe f1d6c8fe-32e2-420a-a051-377e34a8bd8a --format json
```
Start a new quote with this line

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,198 @@
+  },
+  "metadata": {},
+  "name": "compute-uss-1.1.0-134-csm-1.5.x86_64"
+}
```
and end the quote here.

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,198 @@
+
+Or
+
+cray ims images list --format=json | jq '.[] | select(.id == <Image id>)'
```
make this line a separate quote.

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,198 @@
+cray ims images list --format=json | jq '.[] | select(.id == <Image id>)'
+e.g.
+
+# cray ims images list --format=json | jq '.[] | select(.id == "bbe0e9eb-fa8f-4896-9f54-95dbd26de9bb")'                                    
```
And start a new quote before this line.

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
````diff
@@ -0,0 +1,198 @@
+```
+
+Command to untag an IMS image using craycli
+```bash
+cray ims images update < Image id > --metadata-operation remove --metadata-key < key >
````
Again, remove the `bash` so the `<>`s don't get displayed in red, and:

- `< Image id >` -> `<image_id>`
- `< key >` -> `<key>`

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
````diff
@@ -0,0 +1,198 @@
+
+### To boot a single node:
+
+```bash 
+cray bos sessions create --template-name <Bos session template name> --operation reboot --limit < xname of the node>
````
Remove the `bash`

- `<Bos session template name>` -> `<bos_session_template_name>`
- `< xname of the node>` -> `<xname_of_the_node>`

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
````diff
@@ -0,0 +1,198 @@
+
+### To boot all the compute nodes in the given cluster:
+
+```bash
+cray bos sessions create --template-name <BOS session template name> --operation reboot
````
Same thing.  Remove `bash`

- `<BOS session template name>` -> `<bps_session_template_name>`

leliasen-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,198 @@
+# iSCSI SBPS (Scalable Boot Content Projection Service)
+
+iSCSI based content projection solution named Scalable Boot Content Projection service (SBPS) is replacing current DVS (Cray Data Virtualization) to project boot content like 'rootfs' and 'programming environment' (PE) images. SBPS is offering better Reliability, Availability, Security, ease and speed of deployment and ease of management than DVS. Key features of SBPS are as follows:
```
```suggestion
iSCSI based content projection solution named Scalable Boot Content Projection Service (SBPS) is replacing current DVS (Cray Data Virtualization) to project boot content like 'rootfs' and 'programming environment' (PE) images. SBPS is offering better reliability, availability, security, ease and speed of deployment and ease of management than DVS. Key features of SBPS are as follows:

```

aasha-hpe replied:
Done

leliasen-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,198 @@
+1.	Provides open source friendly solution for read-only content projection (rootfs and PE) as it uses LIO (Linux IO)
+2.	Horizontally scalable content projection service (iSCSI target side)
+3.	Delivers active/active IO operation from iSCSI initiator(s) to content projection service
+4.	Delivers seamless failover and failback for iSCSI initiator(s) on iSCSI target(s) or partial network failure
```
```suggestion
4.	Delivers seamless failover and failback for iSCSI initiator(s) on iSCSI target(s) for partial network failure

```
Is this supposed to be `for partial network failure`?

aasha-hpe replied:
No, its or partial network failure only.

aasha-hpe replied:
done.

aasha-hpe commented:
@dborman-hpe Thanks a lot for your review and comments. I have addressed your comments.  There are some checks failing as it says dead links. These links are also under doc review which you and Lindsay have reviewed (Node personalisation, iSCSI metrics and Goss tests). So If no more comments, please merge them.

leliasen-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,222 @@
+saving/clearing the configuration etc as shown in the Figure #1 and the SBPS core service named “SBPS Marshal Agent”runs as
+Linux `systemd` service. The Marshal agent scans IMS and S3 storage for `rootfs` and `PE` images, creates `fileio backing store`
+for the images to be projected. The `rootfs` images to be projected are tagged by BOS (Boot Orchestration Service) when the boot
+of initiator node(s) is triggered. Then the Marshal Agent creates iSCSI `Luns` for each of the `fileio` backing store where the
```
```suggestion
of initiator node(s) is triggered. Then the Marshal Agent creates iSCSI `LUNs` for each of the `fileio` backing store where the

```

aasha-hpe replied:
The word 'LUNs' has to be this way only all caps.

aasha-hpe replied:
Done

leliasen-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,222 @@
+As shown in figure #1 this includes standard Linux kernel, `s3fs` to mount the “boot-images” bucket onto worker node,
+LIO (Linux IO) - an open-source implementation of SCSI target which supports `fileio` backing store, `targetcli` - 
+LIO command-line interface to manage iSCSI devices like creation of `LUNs`, listing of `LUNs`, creation of `fileio backstore`,
+saving/clearing the configuration etc as shown in the Figure #1 and the SBPS core service named “SBPS Marshal Agent”runs as
```
```suggestion
saving/clearing the configuration etc as shown in the Figure #1 and the SBPS core service named “SBPS Marshal Agent” runs as

```

dborman-hpe replied:
Also change the complex double quotes to simple double quotes:
`“SBPS Marshal Agent”runs` -> `"SBPS Marshal Agent" runs`

leliasen-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,222 @@
+iSCSI initiator includes standard Linux kernel, user space iSCSI initiator services and DM (Device Mapper)
+`multipathing` software. DNS SRV and A records are used to discover the target nodes during the boot and
+are part of BOS (Boot orchestration service) session template boot parameters. This BOS session template
+is used to rigger the boot of initiator node(s). The `LUNs` created on the target node which has the
```
```suggestion
is used to trigger the boot of initiator node(s). The `LUNs` created on the target node which has the

```

aasha-hpe replied:
done

leliasen-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,222 @@
+are part of BOS (Boot orchestration service) session template boot parameters. This BOS session template
+is used to rigger the boot of initiator node(s). The `LUNs` created on the target node which has the
+`rootfs`/ `PE` images mapped are thus projected to initiator node(s) when the boot is triggered.
+Basically `rootfs` image projected is used as part of booting the initiator node and `PE` image(s)
```
```suggestion
Basically, the `rootfs` image projected is used as part of booting the initiator node and `PE` image(s)

```

aasha-hpe replied:
done

aasha-hpe replied:
incorporated.

leliasen-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
````diff
@@ -0,0 +1,222 @@
+  o- vhost ............................................................................................................ [Targets: 0]
+  o- xen-pvscsi ....................................................................................................... [Targets: 0]
+```
+In above `targetcli ls` command output, we can see that four `fileio` backing store are created for four `rootfs` images and four iSCSI `LUNs` created which has the `rootfs` id being mapped and ready for `rootfs` image projection.
````
```suggestion
In above `targetcli ls` command output, we can see that four `fileio` backing store are created for four `rootfs` images and four iSCSI `LUNs` are created which has the `rootfs` id being mapped and ready for `rootfs` image projection.

```

aasha-hpe replied:
done

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,242 @@
+# iSCSI SBPS (Scalable Boot Content Projection Service)
+
+iSCSI based boot content projection solution named Scalable Boot Content Projection Service (SBPS)
```
This whole file has `\r\n` line terminators.  Please strip out all the `\r` characters.

dborman-hpe replied:
You can remove the `\r` characters with:
```bash
cat iscsi_sbps.md | tr -d '\r' > x
mv x iscsi_sbps.md
```

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,242 @@
+`programming environment` (PE) images. SBPS is an alternate projection capability for DVS aimed to offer better reliability, availability,
+security, ease and speed of deployment and ease of management than DVS. Key features of SBPS are as follows:
+
+1.	Provides open source friendly solution for read-only content projection (`rootfs` and `PE`) as it uses LIO (Linux IO) which is open source.
+2.	Horizontally scalable content projection service (iSCSI target side)
+3.	Delivers active/active IO operation from iSCSI initiator(s) to content projection service
+4.	Delivers seamless failover and failback for iSCSI initiator(s) on iSCSI target(s) or partial network failure
+5.	Supports projection over HSN (High Speed Network) and NMN (Node Management Network) 
+6.	Does not require additional hardware infrastructure (iSCSI target)
+7.	Co-exists with DVS until it is deprecated (Plan to deprecate DVS in CSM 1.7) 
+8.	Enables future work related to image access control, multi-tenancy, and related zero trust principles 
+9.	Does not require duplication of images from S3
+10.	Supports monitoring of content projection service for performance and reliability engineering
+11.	Aligns with future plans for similar functionality in next generation systems management solutions
+12.	Easy to deploy and manage. 
```
These lines all have hard tabs, replace the hard tabs with a space.  You could also use the the markdown syntax for a numbered list:
```
1. first line
1. second line
1. third line
```

dborman-hpe replied:
Remove trailing whitespace

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,242 @@
+As shown in figure #1, The basic configuration involves two iSCSI target/server (worker node) nodes
+and two iSCSI initiator/client (compute/UAN) nodes connected via HSN and/or NMN where I/O `multipathing`
+is configured. The `rootfs` and `PE` images are hosted in IMS and S3 respectively and both of these 
+images are mapped to “boot-images bucket of S3. DNS SRV and A [DNS - Domain Name Service, SRV – Service,
```
Change `“boot-images` -> `"boot-images"`, that includes using simple double quotes instead of the complex double quote.
Change `SRV – Service` -> `SRV - Service`: a simple dash instead of the complex dash character.

aasha-hpe replied:
am using only the one dash key I have i.e '-', not sure why it is showing complex dash character.

aasha-hpe replied:
> Change `“boot-images` -> `"boot-images"`, that includes using simple double quotes instead of the complex double quote. Change `SRV – Service` -> `SRV - Service`: a simple dash instead of the complex dash character.

this is anyway changed to `boot-images` now.

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,242 @@
+and two iSCSI initiator/client (compute/UAN) nodes connected via HSN and/or NMN where I/O `multipathing`
+is configured. The `rootfs` and `PE` images are hosted in IMS and S3 respectively and both of these 
+images are mapped to “boot-images bucket of S3. DNS SRV and A [DNS - Domain Name Service, SRV – Service,
+ A – Address] records are created and used for target node(s) discovery from initiator node during its boot.
```
Change `A – Address` to `A - Address`: a simple dash instead of the complex dash character.

aasha-hpe replied:
same as above.

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,242 @@
+ A – Address] records are created and used for target node(s) discovery from initiator node during its boot.
+
+iSCSI Target/Server:
+As shown in figure #1 this includes standard Linux kernel, `s3fs` to mount the “boot-images” bucket onto worker node,
```
Change `“boot-images”` -> `"boot-images"` (use simple double quotes)

aasha-hpe replied:
changed to `boo-images` as suggested.

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,242 @@
+2.	Horizontally scalable content projection service (iSCSI target side)
+3.	Delivers active/active IO operation from iSCSI initiator(s) to content projection service
+4.	Delivers seamless failover and failback for iSCSI initiator(s) on iSCSI target(s) or partial network failure
+5.	Supports projection over HSN (High Speed Network) and NMN (Node Management Network) 
```
Remove trailing whitespace

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,242 @@
+5.	Supports projection over HSN (High Speed Network) and NMN (Node Management Network) 
+6.	Does not require additional hardware infrastructure (iSCSI target)
+7.	Co-exists with DVS until it is deprecated (Plan to deprecate DVS in CSM 1.7) 
+8.	Enables future work related to image access control, multi-tenancy, and related zero trust principles 
```
Remove trailing whitespaces from both of these lines

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,242 @@
+
+As shown in figure #1, The basic configuration involves two iSCSI target/server (worker node) nodes
+and two iSCSI initiator/client (compute/UAN) nodes connected via HSN and/or NMN where I/O `multipathing`
+is configured. The `rootfs` and `PE` images are hosted in IMS and S3 respectively and both of these 
```
Remove trailing whitespace

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,242 @@
+
+iSCSI Target/Server:
+As shown in figure #1 this includes standard Linux kernel, `s3fs` to mount the “boot-images” bucket onto worker node,
+LIO (Linux IO) - an open-source implementation of SCSI target which supports `fileio` backing store, `targetcli` - 
```
Remove trailing whitespace

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
````diff
@@ -0,0 +1,242 @@
+```
+e.g.
+```
+# cray ims images list --format=json | jq '.[] | select(.id == "bbe0e9eb-fa8f-4896-9f54-95dbd26de9bb")'                                    
````
Remove trailing whitespace

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,242 @@
+This IMS image tagging is done automatically by BOS in the latest fix which is more efficient than manual way of doing using `craycli`. BOS tags the `rootfs` images when the boot of compute/UAN is triggered. But for untagging an image, we need to use above `craycli` command as untagging of an image is not supported by BOS.
+
+## Creation of BOS session template
+Once the node personalisation is done and GOSS tests are run successfully, create BOS Session Template with SBPS boot parameters. 
```
Remove trailing whitespace

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
````diff
@@ -0,0 +1,242 @@
+
+### To boot a single node:
+
+``` 
````
Remove trailing whitespace

dborman-hpe commented:
- There continue to be multiple lines with trailing whitespace, here's a search from within `vi`:
```
:g/ $/l
12.^IEasy to deploy and manage. $
LIO (Linux IO) - an open-source implementation of SCSI target which supports `fileio` backing store, `targetcli` - $
<b> $
for the details. $
In order to verify the readiness of the iSCSI targets before triggering the boot of compute nodes/ UAN's, we need to run GOSS tests as $
Please refer https://github.com/Cray-HPE/docs-csm/blob/release/1.6/operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_
Nodes_with_SBPS.md for the details $
``` $
```

- The entire file has Windows `\r\n` line terminations, please convert them all to just `\n`
   - I suggest you change your editor setting to use the Unix `\n` line termination instead of the Windows `\r\n` line termination to avoid this issue in the future.

don-bahls-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,432 @@
+* Delivers seamless failover and failback for iSCSI initiator(s) on iSCSI target(s) or partial network failure
+* Supports projection over HSN (High Speed Network) and NMN (Node Management Network)
+* Does not require additional hardware infrastructure (iSCSI target)
+* Co-exists with DVS until it is deprecated (Plan to deprecate DVS in CSM 1.7)
```
Maybe say `(Plan to deprecate DVS for rootfs projection in CSM 1.7)` as DVS is used by customers for purposes other than rootfs projection using non management nodes.   That use will continue to be supported.

aasha-hpe replied:
ok sure

don-bahls-hpe replied:
```suggestion
* Co-exists with DVS until it is deprecated (Plan to deprecate DVS for rootfs projection in CSM 1.7)
```

don-bahls-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,432 @@
+5. Boot compute / UAN's
+6. Monitor iSCSI Metrics
+
+### Node Personalisation (Worker node)
```
We should probably use the US spelling of Personalization rather than the British English spelling (i.e. Personalisation) just to be consistent with other parts of the documentation.

aasha-hpe replied:
Sure :)

dborman-hpe replied:
There are 5 other instances of [Pp]ersonalisation, please also fix them.

aasha-hpe replied:
Done

don-bahls-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,432 @@
+* Horizontally scalable content projection service (iSCSI target side)
+* Delivers active/active IO operation from iSCSI initiator(s) to content projection service
+* Delivers seamless failover and failback for iSCSI initiator(s) on iSCSI target(s) or partial network failure
+* Supports projection over HSN (High Speed Network) and NMN (Node Management Network)
```
It might be nice to say that this is also possible without significant reconfiguration (which separates SBPS from DVS)

aasha-hpe replied:
Did you mean above in additon to the what we are saying in line no. 25?

don-bahls-hpe replied:
```suggestion
* Supports projection over HSN (High Speed Network) and NMN (Node Management Network) without significant reconfiguration.
```

aasha-hpe replied:
Sure.

aasha-hpe replied:
Done

don-bahls-hpe reviewed (approved):
I have a few minor suggestions, but I think this looks reasonable to publish if we don't have time to review those comments.

aasha-hpe commented:
> * There continue to be multiple lines with trailing whitespace, here's a search from within `vi`:
> 
> ```
> :g/ $/l
> 12.^IEasy to deploy and manage. $
> LIO (Linux IO) - an open-source implementation of SCSI target which supports `fileio` backing store, `targetcli` - $
> <b> $
> for the details. $
> In order to verify the readiness of the iSCSI targets before triggering the boot of compute nodes/ UAN's, we need to run GOSS tests as $
> Please refer https://github.com/Cray-HPE/docs-csm/blob/release/1.6/operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_
> Nodes_with_SBPS.md for the details $
> ``` $
> ```
> 
> * The entire file has Windows `\r\n` line terminations, please convert them all to just `\n`
>   
>   * I suggest you change your editor setting to use the Unix `\n` line termination instead of the Windows `\r\n` line termination to avoid this issue in the future.

Taken care.

dborman-hpe commented:
Please make this change to get rid of multi-byte characters:
```
diff --git a/operations/iscsi_sbps/iscsi_sbps.md b/operations/iscsi_sbps/iscsi_sbps.md
index 4f2be37931f..75027dc8897 100644
--- a/operations/iscsi_sbps/iscsi_sbps.md
+++ b/operations/iscsi_sbps/iscsi_sbps.md
@@ -38,11 +38,11 @@ In CSM 1.6, both DVS and SBPS will co-exist and DVS is planned to deprecate in C
 As shown in figure #1, The basic configuration involves two iSCSI target/server (worker node) nodes
 and two iSCSI initiator/client (compute/UAN) nodes connected via HSN and/or NMN where I/O `multipathing`
 is configured. The `rootfs` and `PE` images are hosted in IMS and S3 respectively and both of these
-images are mapped to <E2><80><9C>boot-images bucket of S3. DNS SRV and A [DNS - Domain Name Service, SRV <E2><80><93> Service,
- A <E2><80><93> Address] records are created and used for target node(s) discovery from initiator node during its boot.
+images are mapped to the `boot-images` bucket of S3. DNS SRV and A [DNS - Domain Name Service, SRV - Service,
+ A - Address] records are created and used for target node(s) discovery from initiator node during its boot.
 
 **iSCSI Target/Server:**
-As shown in figure #1 this includes standard Linux kernel, `s3fs` to mount the <E2><80><9C>boot-images<E2><80><9D> bucket onto worker node,
+As shown in figure #1 this includes standard Linux kernel, `s3fs` to mount the `boot-images` bucket onto worker node,
 LIO (Linux IO) - an open-source implementation of SCSI target which supports `fileio` backing store, `targetcli` -
 LIO command-line interface to manage iSCSI devices like creation of `LUNs`, listing of `LUNs`, creation of `fileio backstore`,
 saving/clearing the configuration etc as shown in the Figure #1 and the SBPS core service named "SBPS Marshal Agent" runs as
```
That will resolve my remaining 3 comments from Sept. 12th

dborman-hpe commented:
I've marked as `Resolved` all the comments that I've verified have been addressed in the document, leaving just a few outstanding items.

aasha-hpe commented:
> Please make this change to get rid of multi-byte characters:
> 
> ```
> diff --git a/operations/iscsi_sbps/iscsi_sbps.md b/operations/iscsi_sbps/iscsi_sbps.md
> index 4f2be37931f..75027dc8897 100644
> --- a/operations/iscsi_sbps/iscsi_sbps.md
> +++ b/operations/iscsi_sbps/iscsi_sbps.md
> @@ -38,11 +38,11 @@ In CSM 1.6, both DVS and SBPS will co-exist and DVS is planned to deprecate in C
>  As shown in figure #1, The basic configuration involves two iSCSI target/server (worker node) nodes
>  and two iSCSI initiator/client (compute/UAN) nodes connected via HSN and/or NMN where I/O `multipathing`
>  is configured. The `rootfs` and `PE` images are hosted in IMS and S3 respectively and both of these
> -images are mapped to <E2><80><9C>boot-images bucket of S3. DNS SRV and A [DNS - Domain Name Service, SRV <E2><80><93> Service,
> - A <E2><80><93> Address] records are created and used for target node(s) discovery from initiator node during its boot.
> +images are mapped to the `boot-images` bucket of S3. DNS SRV and A [DNS - Domain Name Service, SRV - Service,
> + A - Address] records are created and used for target node(s) discovery from initiator node during its boot.
>  
>  **iSCSI Target/Server:**
> -As shown in figure #1 this includes standard Linux kernel, `s3fs` to mount the <E2><80><9C>boot-images<E2><80><9D> bucket onto worker node,
> +As shown in figure #1 this includes standard Linux kernel, `s3fs` to mount the `boot-images` bucket onto worker node,
>  LIO (Linux IO) - an open-source implementation of SCSI target which supports `fileio` backing store, `targetcli` -
>  LIO command-line interface to manage iSCSI devices like creation of `LUNs`, listing of `LUNs`, creation of `fileio backstore`,
>  saving/clearing the configuration etc as shown in the Figure #1 and the SBPS core service named "SBPS Marshal Agent" runs as
> ```
> 
> That will resolve my remaining 3 comments from Sept. 12th

I have incorporated this change. Regarding dash, I am using '-' which is the only dash key I see on my keyboard.

aasha-hpe commented:
Thanks a lot @dborman-hpe for all your review and comments/suggestions. I have tried to address them. Pls see if the changes are fine.

dborman-hpe commented:
> > Please make this change to get rid of multi-byte characters:
> > ```
> > diff --git a/operations/iscsi_sbps/iscsi_sbps.md b/operations/iscsi_sbps/iscsi_sbps.md
> > index 4f2be37931f..75027dc8897 100644
> > --- a/operations/iscsi_sbps/iscsi_sbps.md
> > +++ b/operations/iscsi_sbps/iscsi_sbps.md
> > @@ -38,11 +38,11 @@ In CSM 1.6, both DVS and SBPS will co-exist and DVS is planned to deprecate in C
> >  As shown in figure #1, The basic configuration involves two iSCSI target/server (worker node) nodes
> >  and two iSCSI initiator/client (compute/UAN) nodes connected via HSN and/or NMN where I/O `multipathing`
> >  is configured. The `rootfs` and `PE` images are hosted in IMS and S3 respectively and both of these
> > -images are mapped to <E2><80><9C>boot-images bucket of S3. DNS SRV and A [DNS - Domain Name Service, SRV <E2><80><93> Service,
> > - A <E2><80><93> Address] records are created and used for target node(s) discovery from initiator node during its boot.
> > +images are mapped to the `boot-images` bucket of S3. DNS SRV and A [DNS - Domain Name Service, SRV - Service,
> > + A - Address] records are created and used for target node(s) discovery from initiator node during its boot.
> >  
> >  **iSCSI Target/Server:**
> > -As shown in figure #1 this includes standard Linux kernel, `s3fs` to mount the <E2><80><9C>boot-images<E2><80><9D> bucket onto worker node,
> > +As shown in figure #1 this includes standard Linux kernel, `s3fs` to mount the `boot-images` bucket onto worker node,
> >  LIO (Linux IO) - an open-source implementation of SCSI target which supports `fileio` backing store, `targetcli` -
> >  LIO command-line interface to manage iSCSI devices like creation of `LUNs`, listing of `LUNs`, creation of `fileio backstore`,
> >  saving/clearing the configuration etc as shown in the Figure #1 and the SBPS core service named "SBPS Marshal Agent" runs as
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > That will resolve my remaining 3 comments from Sept. 12th
> 
> I have incorporated this change. Regarding dash, I am using '-' which is the only dash key I see on my keyboard.

Well, you did manage to fix it.  The 3 character `<E2><80><93>` sequences are now just `-`.

dborman-hpe reviewed (approved):
LGTM

dborman-hpe commented on operations/iscsi_sbps/iscsi_sbps.md:
```diff
@@ -0,0 +1,432 @@
+* Uninstall CPS
+* Enable iSCSI SBPS
+
+To disable DVS please refer:
+https://github.hpe.com/hpe/hpc-shasta-os-uss-docs/blob/master/dac-assets/documents/operations/DVS_and_SBPS.md
+
+To uninstall CPS services please refer:
+https://github.hpe.com/hpe/hpc-shasta-os-uss-docs/blob/master/dac-assets/documents/operations/Uninstall_CPS.md
+
+To enable iSCSI SBPS please refer:
+https://github.hpe.com/hpe/hpc-shasta-os-uss-docs/blob/master/dac-assets/documents/operations/Scalable_Boot_projection_service.md
```
These bare URLs need to be fixed, see https://github.com/updownpress/markdown-lint/blob/master/rules/034-no-bare-urls.md
```
diff --git a/operations/iscsi_sbps/iscsi_sbps.md b/operations/iscsi_sbps/iscsi_sbps.md
index 3fc398772d8..f75a2401316 100644
--- a/operations/iscsi_sbps/iscsi_sbps.md
+++ b/operations/iscsi_sbps/iscsi_sbps.md
@@ -405,13 +405,13 @@ Please refer [iSCSI Metrics](https://github.com/Cray-HPE/sbps-marshal/blob/main/
 * Enable iSCSI SBPS
 
 To disable DVS please refer:
-https://github.hpe.com/hpe/hpc-shasta-os-uss-docs/blob/master/dac-assets/documents/operations/DVS_and_SBPS.md
+<https://github.hpe.com/hpe/hpc-shasta-os-uss-docs/blob/master/dac-assets/documents/operations/DVS_and_SBPS.md>
 
 To uninstall CPS services please refer:
-https://github.hpe.com/hpe/hpc-shasta-os-uss-docs/blob/master/dac-assets/documents/operations/Uninstall_CPS.md
+<https://github.hpe.com/hpe/hpc-shasta-os-uss-docs/blob/master/dac-assets/documents/operations/Uninstall_CPS.md>
 
 To enable iSCSI SBPS please refer:
-https://github.hpe.com/hpe/hpc-shasta-os-uss-docs/blob/master/dac-assets/documents/operations/Scalable_Boot_projection_service.md
+<https://github.hpe.com/hpe/hpc-shasta-os-uss-docs/blob/master/dac-assets/documents/operations/Scalable_Boot_projection_service.md>
 
 Note: These URL references will be replaced with appropriate document name and number in support center.
```
